packages/fiori/src/ShellBar.ts

```diff
@@ -693,35 +756,34 @@ class ShellBar extends UI5Element {
 					order: shouldStayOnScreen ? 1 : -1,
 				},
 			};
-		});
+		}) as Array<IShelBarItemInfo>;
 
 		this._updateItemsInfo(newItems);
 	}
 
 	_handleActionsOverflow() {
-		const rightContainerRect = this.shadowRoot.querySelector(".ui5-shellbar-overflow-container-right").getBoundingClientRect();
+		const rightContainerRect = this.shadowRoot!.querySelector(".ui5-shellbar-overflow-container-right")!.getBoundingClientRect();
 		let overflowSelector = ".ui5-shellbar-button:not(.ui5-shellbar-overflow-button):not(.ui5-shellbar-invisible-button)";
 
 		if (this.showSearchField) {
 			overflowSelector += ",.ui5-shellbar-search-field";
 		}
 
-		const elementsToOverflow = this.shadowRoot.querySelectorAll(overflowSelector);
+		const elementsToOverflow = this.shadowRoot!.querySelectorAll(overflowSelector);
 		const isRTL = this.effectiveDir === "rtl";
 
-		let overflowCount = [].filter.call(elementsToOverflow, icon => {
-			const iconRect = icon.getBoundingClientRect();
+		const overflowCount = [].filter.call(elementsToOverflow, icon => {
```

Comment:
change to:

```ts
		const elementsToOverflow = this.shadowRoot!.querySelectorAll<Button>(overflowSelector);
		const isRTL = this.effectiveDir === "rtl";

		const overflowCount = [...elementsToOverflow].filter(icon => {....
```
If we use generics of querySelectorAll it will allow us to not use casting for icon as button